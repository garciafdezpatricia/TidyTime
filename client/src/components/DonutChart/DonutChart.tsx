import React from 'react';

export default function DonutChart({ total, value }: {total:any, value:any}) {
  const radius = 50; // radio del c√≠rculo
  const circumference = 2 * Math.PI * radius;
  const strokeDashoffset = total !== 0 ? circumference - (value / total) * circumference : 0;

  return (
    <svg width="20%" height="20%" viewBox="0 0 120 120">
      <circle
        cx="60"
        cy="60"
        r={radius}
        stroke="#d8d8d8"
        strokeWidth="10"
        fill="transparent"
      />
      <circle
        cx="60"
        cy="60"
        r={radius}
        stroke="#000000"
        strokeWidth="10"
        fill="transparent"
        strokeDasharray={circumference}
        strokeDashoffset={strokeDashoffset}
        transform={`rotate(-90 60 60)`}
        style={{ transition: 'stroke-dashoffset 0.3s ease-out' }}
      />
      <text
        x="60"
        y="60"
        textAnchor="middle"
        dy=".3em"
        fontSize="20">
        {`${value}/${total}`}
      </text>
    </svg>
  );
};
