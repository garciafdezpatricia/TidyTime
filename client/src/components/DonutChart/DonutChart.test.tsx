import { render } from "@testing-library/react";
import '@testing-library/jest-dom';
import DonutChart from '@/src/components/DonutChart/DonutChart';

describe('DonutChart', () => {
    it('renders without crashing', () => {
      const { container } = render(<DonutChart total={2} value={0} />);
      expect(container).toBeInTheDocument();
    });

    it('matches snapshot', () => {
        const { asFragment } = render(<DonutChart total={2} value={0} />);
        expect(asFragment()).toMatchSnapshot();
      });
  });