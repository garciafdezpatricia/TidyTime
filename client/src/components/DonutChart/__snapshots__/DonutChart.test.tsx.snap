// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DonutChart matches snapshot 1`] = `
<DocumentFragment>
  <svg
    height="20%"
    viewBox="0 0 120 120"
    width="20%"
  >
    <circle
      cx="60"
      cy="60"
      fill="transparent"
      r="50"
      stroke="#d8d8d8"
      stroke-width="10"
    />
    <circle
      cx="60"
      cy="60"
      fill="transparent"
      r="50"
      stroke="#000000"
      stroke-dasharray="314.1592653589793"
      stroke-dashoffset="314.1592653589793"
      stroke-width="10"
      style="transition: stroke-dashoffset 0.3s ease-out;"
      transform="rotate(-90 60 60)"
    />
    <text
      data-testid="total-tasks-donut"
      dy=".3em"
      font-size="20"
      text-anchor="middle"
      x="60"
      y="60"
    >
      0/2
    </text>
  </svg>
</DocumentFragment>
`;
